drivers/sensors/cirque_pinnacle.h

```diff
@@ -21,24 +21,29 @@
 #    define CIRQUE_PINNACLE_DIAMETER_MM 40
 #endif
 
+#if CIRQUE_PINNACLE_POSITION_MODE
 // Coordinate scaling values
-#ifndef CIRQUE_PINNACLE_X_LOWER
-#    define CIRQUE_PINNACLE_X_LOWER 127 // min "reachable" X value
-#endif
-#ifndef CIRQUE_PINNACLE_X_UPPER
-#    define CIRQUE_PINNACLE_X_UPPER 1919 // max "reachable" X value
-#endif
-#ifndef CIRQUE_PINNACLE_Y_LOWER
-#    define CIRQUE_PINNACLE_Y_LOWER 63 // min "reachable" Y value
-#endif
-#ifndef CIRQUE_PINNACLE_Y_UPPER
-#    define CIRQUE_PINNACLE_Y_UPPER 1471 // max "reachable" Y value
-#endif
-#ifndef CIRQUE_PINNACLE_X_RANGE
-#    define CIRQUE_PINNACLE_X_RANGE (CIRQUE_PINNACLE_X_UPPER - CIRQUE_PINNACLE_X_LOWER)
-#endif
-#ifndef CIRQUE_PINNACLE_Y_RANGE
-#    define CIRQUE_PINNACLE_Y_RANGE (CIRQUE_PINNACLE_Y_UPPER - CIRQUE_PINNACLE_Y_LOWER)
+#    ifndef CIRQUE_PINNACLE_X_LOWER
+#        define CIRQUE_PINNACLE_X_LOWER 127 // min "reachable" X value
+#    endif
+#    ifndef CIRQUE_PINNACLE_X_UPPER
+#        define CIRQUE_PINNACLE_X_UPPER 1919 // max "reachable" X value
+#    endif
+#    ifndef CIRQUE_PINNACLE_Y_LOWER
+#        define CIRQUE_PINNACLE_Y_LOWER 63 // min "reachable" Y value
+#    endif
+#    ifndef CIRQUE_PINNACLE_Y_UPPER
+#        define CIRQUE_PINNACLE_Y_UPPER 1471 // max "reachable" Y value
+#    endif
+#    ifndef CIRQUE_PINNACLE_X_RANGE
+#        define CIRQUE_PINNACLE_X_RANGE (CIRQUE_PINNACLE_X_UPPER - CIRQUE_PINNACLE_X_LOWER)
+#    endif
+#    ifndef CIRQUE_PINNACLE_Y_RANGE
+#        define CIRQUE_PINNACLE_Y_RANGE (CIRQUE_PINNACLE_Y_UPPER - CIRQUE_PINNACLE_Y_LOWER)
+#    endif
+#else
+#    define CIRQUE_PINNACLE_X_RANGE 256
+#    define CIRQUE_PINNACLE_Y_RANGE 256
```

Comment:
Does this line up with the 
> Resolution 245 x 1/2 cpi

mentioned under PS/2 mode? Or is CPI intended to be an arbitrary scale for relative mode?

I was thinking `cirque_pinnacle_get_cpi()` and `cirque_pinnacle_set_cpi()` need to be overridden for relative mode.

Reply:
Scale is arbitrary, there is no mention of how this is scaled in datasheet. The scale is probably the same as in absolute mode, however the range is not.

Reply:
I was messing around with that _245 x 1/2_ number from the PS/2 portion of the spec and can't get it to make sense....
It's not the same as absolute mode either from what I can tell. I give up.

Do you want to just do something like this in _pointing_device_drivers.c_:
```
#    if CIRQUE_PINNACLE_POSITION_MODE
uint16_t cirque_pinnacle_get_cpi(void) {
    return CIRQUE_PINNACLE_PX_TO_INCH(cirque_pinnacle_get_scale());
}
void cirque_pinnacle_set_cpi(uint16_t cpi) {
    cirque_pinnacle_set_scale(CIRQUE_PINNACLE_INCH_TO_PX(cpi));
}
#    endif

// clang-format off
const pointing_device_driver_t pointing_device_driver = {
    .init       = cirque_pinnacle_init,
    .get_report = cirque_pinnacle_get_report,
#    if CIRQUE_PINNACLE_POSITION_MODE
    .set_cpi    = cirque_pinnacle_set_cpi,
    .get_cpi    = cirque_pinnacle_get_cpi,
#    else
    .set_cpi    = cirque_pinnacle_set_scale,
    .get_cpi    = cirque_pinnacle_get_scale
#    endif
};
```
Empirically it gives sensitivity in a similar ballpark to the absolute mode scaling, and saves a bit more space.
Maybe split the relative mode and absolute mode driver into two sections divided by one `#if CIRQUE_PINNACLE_POSITION_MODE`, with a separate much more concise definition of `cirque_pinnacle_get_report()` for relative mode?

Reply:
On PS/2 mode trackpad has a `Set Resolution` and `Set Scaling` commands. I think the `245 x 1/2 cpi` is somehow relative to the values set there. Maybe they are available in extended registers, but there is zero documentation for them.